crates/unftp-auth-jsonfile/src/lib.rs

```diff
@@ -208,6 +218,37 @@ struct UserCreds {
     pub password: Password,
     pub client_cert: Option<ClientCertCredential>,
     pub allowed_ip_ranges: Option<IpRange<Ipv4Net>>,
+    pub home: Option<PathBuf>,
+}
+
+#[derive(Debug)]
+pub struct User {
+    username: String,
+    home: Option<PathBuf>,
```

Comment:
So you're ok with combining authentication and home directories in one module, just not in unftp-auth-jsonfile itself?  Yes, I can copy all of the relevant parts directly into the cap-ftpd-worker.

Reply:
I actually want to introduce a new concept here in libunftp, that of user config providers. So that we can have crates like `unftp-usr-sql` that in this case will store the user information in a SQL database while still allowing to authenticate in a different way.

Reply:
What would you like to do for now, before user config providers are ready?

Reply:
This will suffice:

> Yes, I can copy all of the relevant parts directly into the cap-ftpd-worker.

leaving `unftp-auth-jsonfile` largely untouched.